### PR TITLE
Extend XPT2046 touch driver to add temperature, batt voltage, aux input voltage read support

### PR DESCRIPTION
**Pull request by dvosully**
The XPT2046 touch controller chip (and variants) supports reading temperature, battery voltage and auxiliary input voltage.

This patch extends the XPT2046 driver with functions to read these values. Tested with the TSC2046 chip (equivalent).

### CODE REVIEW

File: components/display/touch_panel/xpt2046/xpt2046.c [hunk 1]
```diff
@@ -272,3 +278,62 @@ esp_err_t xpt2046_calibration_run(const scr_driver_t *screen, bool recalibrate)
...
+    ret = xpt2046_get_sample(XPT2046_TEMP0_CMD, &temp0);
+    TOUCH_CHECK(ESP_OK == ret, "Temp0 read failed", ESP_FAIL);
+
+    // 12 bit = 4095 counts. 2.507V full scale internal reference. 0.0021V/degC characteristic.
+    //  599.5mV @25degC nominal
+    *temperature = (float) (XPT2046_TEMP0_COUNTS_AT_25C - temp0) * (2.507 / 4095.0) / 0.0021 + 25.0;
+
+    // Last reading is to turn off the Vref
+    ret = xpt2046_get_sample(XPT2046_TOUCH_CMD_Z1, &temp0);
+
```
Comment (Lzw655) [inline components/display/touch_panel/xpt2046/xpt2046.c:299]:
```suggestion
    TOUCH_CHECK(ESP_OK == ret, "Vref turn off failed", ESP_FAIL);

```

File: components/display/touch_panel/xpt2046/xpt2046.c [hunk 2]
```diff
@@ -272,3 +278,62 @@ esp_err_t xpt2046_calibration_run(const scr_driver_t *screen, bool recalibrate)
...
+    ret = xpt2046_get_sample(XPT2046_VBAT_CMD, &vbat);
+    // Second reading is to get the result
+    ret = xpt2046_get_sample(XPT2046_VBAT_CMD, &vbat);
+    TOUCH_CHECK(ESP_OK == ret, "Vbat read failed", ESP_FAIL);
+
+    *voltage = (float) vbat * (2.507 / 4095.0) * 4.0;
+
+    // Last reading is to turn off the Vref
+    ret = xpt2046_get_sample(XPT2046_TOUCH_CMD_Z1, &vbat);
+
```
Comment (Lzw655) [inline components/display/touch_panel/xpt2046/xpt2046.c:318]:
```suggestion
    TOUCH_CHECK(ESP_OK == ret, "Vref turn off failed", ESP_FAIL);
    
```

File: components/display/touch_panel/xpt2046/xpt2046.c [hunk 3]
```diff
@@ -272,3 +278,62 @@ esp_err_t xpt2046_calibration_run(const scr_driver_t *screen, bool recalibrate)
...
+    ret = xpt2046_get_sample(XPT2046_AUXIN_CMD, &vin);
+    // Second reading is to get the result
+    ret = xpt2046_get_sample(XPT2046_AUXIN_CMD, &vin);
+    TOUCH_CHECK(ESP_OK == ret, "Aux read failed", ESP_FAIL);
+
+    *voltage = (float) vin * (2.507 / 4095.0);
+
+    // Last reading is to turn off the Vref
+    ret = xpt2046_get_sample(XPT2046_TOUCH_CMD_Z1, &vin);
+
```
Comment (Lzw655) [inline components/display/touch_panel/xpt2046/xpt2046.c:337]:
```suggestion
    TOUCH_CHECK(ESP_OK == ret, "Vref turn off failed", ESP_FAIL);
    
```

File: components/display/touch_panel/xpt2046/xpt2046.h [hunk 4]
```diff
@@ -99,6 +99,39 @@ esp_err_t xpt2046_calibration_run(const scr_driver_t *screen, bool recalibrate);
...
+/**
+ * @brief Get the temperature from the touchscreen controller
+ *
+ * @param temperature a pointer to a float to store the temperature in degrees C
+ *
+ * @return
+ *     - ESP_OK Success
+ *     - ESP_FAIL Fail
+ */
+esp_err_t xpt2046_get_aux_v(float* temperature);
```
Comment (Lzw655) [inline components/display/touch_panel/xpt2046/xpt2046.h:133]: Please modify the descriptions and the name of argument for these two APIs.

**Review by Lzw655 — CHANGES_REQUESTED** [review]
@dvosully Hi, It's very nice to add three extra functions for XPT2046. Thanks for your contributions! And I left some suggestions, PTAL.

**Comment by dvosully** [comment]
Thanks, I have incorporated your suggested improvements.

**Review by Lzw655 — APPROVED** [review]
Nice work, can be merged!

**Pull request closed without merging**